src/Snake.DataAccess/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,80 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Snake.Game.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+
+namespace Snake.DataAccess.Repositories
+{
+    public class GenericRepository<TEntity>: IGenericRepository<TEntity> where TEntity : class
+    {
+        internal SnakeDBContext context;
+        internal DbSet<TEntity> dbSet;
+
+        public GenericRepository(SnakeDBContext context)
+        {
+            this.context = context;
+            this.dbSet = context.Set<TEntity>();
+        }
+
+        public virtual IEnumerable<TEntity> Get(
+            Expression<Func<TEntity, bool>> filter = null,
+            Func<IQueryable<TEntity>, IOrderedQueryable<TEntity>> orderBy = null,
+            string includeProperties = "")
+        {
+            IQueryable<TEntity> query = dbSet;
+
+            if (filter != null)
+            {
+                query = query.Where(filter);
+            }
+
+            foreach (var includeProperty in includeProperties.Split
+                (new char[] { ',' }, StringSplitOptions.RemoveEmptyEntries))
+            {
+                query = query.Include(includeProperty);
+            }
+
+            if (orderBy != null)
+            {
+                return orderBy(query).ToList();
+            }
+            else
+            {
+                return query.ToList();
+            }
+        }
+
+        public virtual TEntity GetByID(object id)
+        {
+            return dbSet.Find(id);
+        }
+
+        public virtual void Insert(TEntity entity)
+        {
+            dbSet.Add(entity);
+        }
+
+        public virtual void Delete(object id)
```

Comment:
This method is not and implementation of an interface, thus it probably should be protected or private.